pkg/apis/core/v1/defaults.go

```diff
@@ -349,6 +350,21 @@ func SetDefaults_HTTPGetAction(obj *v1.HTTPGetAction) {
 		obj.Scheme = v1.URISchemeHTTP
 	}
 }
+
+// SetDefaults_Namespace adds a default label for all namespaces
+func SetDefaults_Namespace(obj *v1.Namespace) {
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.DefaultNamespaceLabelStamping) {
+		if obj.Labels == nil {
+			obj.Labels = map[string]string{}
+		}
+		// Default label that matches the name of the namespace, for selecting w/o apriori knowledge of labels.
```

Comment:
it's notable this is overriding any existing value... probably worth a comment that this is intentional